ipfsspec/core.py

```diff
@@ -166,7 +174,23 @@ def ls(self, path, detail=True, **kwargs):
         logger.debug("ls on %s", path)
         res = self._gw_apipost("ls", arg=path)
         links = res["Objects"][0]["Links"]
-        types = {1: "directory", 2: "file"}
+
+        # Documentation for the type codes doesn't seem to have a great location.
+        # Currently 2023-10-17, the ls API docs specify the ls return
+        # structure, but not what the type integers mean:
+        # http://docs.ipfs.tech.ipns.localhost:8080/reference/kubo/rpc/#api-v0-ls
+        # Some information about type codes can be found here:
+        # https://ipfs-search.readthedocs.io/en/latest/ipfs_datatypes.html
+        # https://github.com/ipfs/go-unixfs/blob/master/pb/unixfs.proto
+
+        types = {
+            0: "raw",
+            1: "directory",
+            2: "file",
+            3: "metadata",
+            4: "symlink",
+            5: "shard",
+        }
```

Comment:
This will also touch what the generic `fsspec` API expects as a return form `ls` (and `info`). I don't know a definitve answer here, but maybe @martindurant could chime in. If I look at e.g. the [local filesystem implementation](https://github.com/fsspec/filesystem_spec/blob/3fbb6fee33b46cccb015607630843dea049d3243/fsspec/implementations/local.py#L93), there are only types `file`, `directory` and `other`. There's additionally the `islink` flag.
So @martindurant, would `raw`, `metadata`, `symlink` and `shard` be valid return values for the `type`? And more specifically, what would be the best return value to communicate the presence of a symlink?